src/core_modules/capture-core/rules/postProcessRulesEffects.js

```diff
@@ -27,14 +38,13 @@ const postProcessAssignEffects = ({
 }) => {
     const optionSets = foundation.getElements().filter(({ optionSet }) => optionSet).reduce((acc, { id, optionSet }) => {
         // $FlowFixMe
-        acc[id] = optionSet.options;
+        acc[id] = { options: optionSet.options, dataElementType: optionSet.dataElement.type };
         return acc;
     }, {});
 
     // If a value gets assigned to an option set it must match one of its available options
     assignValueEffects.map((effect) => {
-        // Using == because effect.value is always a string whereas option.value can be a number
-        if (optionSets[effect.id] && !optionSets[effect.id].some(option => option.value == effect.value)) {
+        if (!isValidOptionSet(optionSets[effect.id], effect.value)) {
```

Comment:
We should check that `effect.id` is an option set before calling `isValidOptionSet`:

`if (optionSets[effect.id] && !isValidOptionSet(optionSets[effect.id], effect.value))`

Otherwise the example program rule in the jira blanks the first name field instead of assigning 'Test'.

Reply:
Good catch! I updated the code with the appropriate checks. Thanks!